src/entities.rs

```diff
@@ -236,6 +236,22 @@ impl Entities {
         Ok(meta.location)
     }
 
+    /// Returns `None` if the given id would represent an index outside of `meta`.
+    pub unsafe fn resolve_unknown_gen(&self, id: u32) -> Option<Entity> {
+        let meta_len = self.meta.len();
+        if meta_len + self.pending.load(Ordering::Relaxed) as usize <= id as usize {
+            None
```

Comment:
Since such an `id` *definitely* violates the safety constraint, let's just panic here and have a simpler signature which does not give the impression of having well-defined error conditions.

Reply:
Changed as requested; please see latest commit.